**Problem:** Write a multiple regression program for predicting CO2 emission level when volume 1300 of Car is and Engine weight is 3300kg.
    
Refer Car Dataset  

Multiple regression is like linear regression, but with more than one independent value, meaning that we try to predict a value based on two or more variables.

In [1]:
import pandas

In [2]:
df = pandas.read_excel("Car_data.xlsx")
df.head(5)

Volume  Weight  CO2
0    1000     790   99
1    1200    1160   95
2    1000     929   95
3     900     865   90
4    1500    1140  105

In [3]:
X = df[['Weight', 'Volume']]
y = df['CO2']

In [4]:
from sklearn import linear_model

In [5]:
regr = linear_model.LinearRegression()
regr.fit(X.values, y)

LinearRegression()

In [6]:
#predict the CO2 emission of a car where the weight is 2300kg, and the volume is 1300cm3:
predictedCO2 = regr.predict([[2300, 1300]])

In [7]:
print(predictedCO2)

[107.2087328]


# Coefficient

The coefficient is a factor that describes the relationship with an unknown variable.

Example: if x is a variable, then 2x is x two times. x is the unknown variable, and the number 2 is the coefficient.

In this case, we can ask for the coefficient value of weight against CO2, and for volume against CO2. The answer(s) we get tells us what would happen if we increase, or decrease, one of the independent values.

In [8]:
print(regr.coef_)

[0.00755095 0.00780526]


The result array represents the coefficient values of weight and volume.

Weight: 0.00755095

Volume: 0.00780526

These values tell us that if the weight increase by 1kg, the CO2 emission increases by 0.00755095g.

And if the engine size (Volume) increases by 1 cm3, the CO2 emission increases by 0.00780526 g.

I think that is a fair guess, but let test it!

We have already predicted that if a car with a 1300cm3 engine weighs 2300kg, the CO2 emission will be approximately 107g.

What if we increase the weight with 1000kg?

In [9]:
predictedCO2 = regr.predict([[3300, 1300]])

print(predictedCO2)

[114.75968007]


We have predicted that a car with 1.3 liter engine, and a weight of 3300 kg, will release approximately 115 grams of CO2 for every kilometer it drives.

Which shows that the coefficient of 0.00755095 is correct:

107.2087328 + (1000 * 0.00755095) = 114.75968